tianshou/utils/net/common.py

```diff
@@ -9,35 +10,74 @@ class Net(nn.Module):
     """Simple MLP backbone. For advanced usage (how to customize the network),
     please refer to :ref:`build_the_network`.
 
-    :param concat: whether the input shape is concatenated by state_shape
+    :param bool concat: whether the input shape is concatenated by state_shape
         and action_shape. If it is True, ``action_shape`` is not the output
         shape, but affects the input shape.
+    :param bool dueling: whether to use dueling network to calculate Q values
+        (for Dueling DQN), defaults to False.
+    :param bool layer_norm: whether to use layer norm before ReLU, defaults to
+        False.
     """
 
-    def __init__(self, layer_num, state_shape, action_shape=0, device='cpu',
-                 softmax=False, concat=False, hidden_layer_size=128):
+    def __init__(self, layer_num: int, state_shape: tuple,
+                 action_shape: Optional[tuple] = 0,
+                 device: Union[str, torch.device] = 'cpu',
+                 softmax: bool = False, concat: bool = False,
+                 hidden_layer_size: int = 128,
+                 dueling: Tuple[int, int] = None, layer_norm: bool = False):
         super().__init__()
         self.device = device
+        self.dueling = dueling
+        self.softmax = softmax
         input_size = np.prod(state_shape)
         if concat:
             input_size += np.prod(action_shape)
-        self.model = [
-            nn.Linear(input_size, hidden_layer_size),
-            nn.ReLU(inplace=True)]
+
+        def miniblock(inp, oup):
+            ret = [nn.Linear(inp, oup)]
+            if layer_norm:
+                ret += [nn.LayerNorm(oup)]
+            ret += [nn.ReLU(inplace=True)]
+            return ret
```

Comment:
It would be better to have a static function rather than a function inside a function. Static functions can be reused elsewhere, but inner functions cannot.

Reply:
But this method is only used in the init.

Reply:
> But this method is only used in the init.

Torchvision can be a reference. Their ResNet has a ``make_layer`` function. Even though it is just used during init, who knows if a user would like to access it ┓( ´∀` )┏